internal/api/mfa.go

```diff
@@ -89,13 +89,35 @@ func (a *API) enrollPhoneFactor(w http.ResponseWriter, r *http.Request, params *
 	if err := models.DeleteExpiredFactors(db, config.MFA.FactorExpiryDuration); err != nil {
 		return err
 	}
+	var factorsToDelete []models.Factor
+	for _, factor := range user.Factors {
+		switch {
+		case factor.FriendlyName == params.FriendlyName:
+			return unprocessableEntityError(
+				ErrorCodeMFAFactorNameConflict,
+				fmt.Sprintf("A factor with the friendly name %q for this user already exists", factor.FriendlyName),
+			)
+
+		case factor.IsPhoneFactor():
+			if factor.IsVerified() {
+				return unprocessableEntityError(
+					ErrorCodeVerifiedFactorExists,
+					"A verified phone factor already exists, unenroll the existing factor to continue",
+				)
+			}
```

Comment:
Why is this limitation? I should be able to enroll multiple phone numbers as MFA factors.

Reply:
Sounds good, let's do this. Initial consideration was that most users might not need more than one number but it's better to allow the general case like with enterprises where they might have multiple phones for testing etc etc